From b1857a2836c9a91ef4e359ef7ba949b54c77219d Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Mon, 20 Jan 2025 15:07:19 +0900
Subject: [PATCH] gpg: Fix handling with no CRC armor.

* g10/armor.c (radix64_read): Set ->any_data if any data is available.

--

GnuPG-bug-id: 7071
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 g10/armor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/g10/armor.c b/g10/armor.c
index 948b8c802..036b72772 100644
--- a/g10/armor.c
+++ b/g10/armor.c
@@ -1054,11 +1054,13 @@ radix64_read( armor_filter_context_t *afx, IOBUF a, size_t *retn,
     afx->radbuf[0] = val;
 
     if( n )
-      gcry_md_write (afx->crc_md, buf, n);
+      {
+        gcry_md_write (afx->crc_md, buf, n);
+        afx->any_data = 1;
+      }
 
     if( checkcrc ) {
 	gcry_md_final (afx->crc_md);
-	afx->any_data = 1;
 	afx->inp_checked=0;
 	afx->faked = 0;
 	for(;;) { /* skip lf and pad characters */
-- 
2.49.0

